yoga/Yoga.cpp

```diff
@@ -2953,9 +2953,9 @@ static void YGNodelayoutImpl(
       } else {
         if (!node->getConfig()->useLegacyStretchBehaviour &&
             ((YGFloatIsUndefined(
-                  collectedFlexItemsValues.totalFlexGrowFactors) &&
+                  collectedFlexItemsValues.totalFlexGrowFactors) ||
               collectedFlexItemsValues.totalFlexGrowFactors == 0) ||
-             (YGFloatIsUndefined(node->resolveFlexGrow()) &&
+             (YGFloatIsUndefined(node->resolveFlexGrow()) ||
```

Comment:
Guess those extra parentheses are redundant, now that it's a list of OR clauses... granted, the expression is quite hard to unpack visually, and extra parentheses don't help here, quite an opposite

Reply:
I actually believe that the initial intent there may have been to have it like this instead:
```
if (!node->getConfig()->useLegacyStretchBehaviour &&
            ((!YGFloatIsUndefined(
                  collectedFlexItemsValues.totalFlexGrowFactors) &&
              collectedFlexItemsValues.totalFlexGrowFactors == 0) ||
             (!YGFloatIsUndefined(node->resolveFlexGrow()) &&
              node->resolveFlexGrow() == 0))) {
```

Reply:
Yeah, you're right. I had update this code.